pkg/iommu/iommu.go

```diff
@@ -0,0 +1,59 @@
+package iommu
+
+import (
+	"fmt"
+	"io/ioutil"
+	"strconv"
+	"strings"
+)
+
+func GroupMapForPCIDevices(groupPaths []string) map[string]int {
+	groupMap := make(map[string]int)
+	for _, groupPath := range groupPaths {
+		address := strings.Split(groupPath, "/")[6]
+		group := GroupForPCIDevice(address, groupPaths)
+		groupMap[address] = group
+	}
+	return groupMap
+}
+
+// return the iommu group of the PCI device
+func GroupForPCIDevice(address string, groupPaths []string) int {
+	for _, groupPath := range groupPaths {
+		if strings.HasSuffix(groupPath, address) {
+			// extract iommu group from path and return
+			iommuGroupStr := strings.Split(groupPath, "/")[4]
+			iommuGroup, err := strconv.Atoi(iommuGroupStr)
+			if err != nil {
```

Comment:
we should change method signature to return error and let reconcile loop handle the same